src/nft/hooks/useFetchAssets.ts

```diff
@@ -100,3 +106,48 @@ export function useFetchAssets(): () => Promise<void> {
     tokenTradeInput,
   ])
 }
+
+export const useFetchSingleAsset = () => {
+  const { account } = useWeb3React()
+  const [fetchGqlRoute] = useNftRouteLazyQuery()
+  const purchaseAssets = usePurchaseAssets()
+
+  const [isLoading, setIsLoading] = useState(false)
+
+  const fetchAndPurchaseSingleAsset = useCallback(
+    async (asset: GenieAsset) => {
+      setIsLoading(true)
+
+      fetchGqlRoute({
+        variables: {
+          senderAddress: account ? account : '',
+          nftTrades: buildNftTradeInput([asset]),
+          tokenTrades: undefined,
+        },
+        pollInterval: 0,
+        fetchPolicy: 'no-cache',
+        onCompleted: (data) => {
+          setIsLoading(false)
+
+          if (!data.nftRoute || !data.nftRoute.route) {
+            return
+          }
+
+          const { route, routeResponse } = buildRouteResponse(data.nftRoute, false)
+          const { hasPriceAdjustment, updatedAssets } = compareAssetsWithTransactionRoute([asset], route)
+          const { unavailable } = filterUpdatedAssetsByState(updatedAssets)
+          const invalidData = hasPriceAdjustment || unavailable.length > 0
```

Comment:
Can you rebase off main so we can test with pooled assets?

Reply:
I still can't seem to buy pooled assets even without a price change. Tested with this [collection](https://interface-git-jack-singleassetbuy-uniswap.vercel.app/#/nfts/collection/0xec9c519d49856fd2f8133a0741b4dbe002ce211b)
![pooledBuy](https://github.com/Uniswap/interface/assets/11512321/114a7e67-e2d7-4b3e-9acf-b62a10e85f9f)